pkg/stanza/flush/flush.go

```diff
@@ -8,6 +8,8 @@ import (
 	"time"
 )
 
+var Now = time.Now
```

Comment:
Does it need to be exported?

Reply:
This belongs to the package `flush` and if it's not exported it won't be possible to be set in `TestFlushPeriodEOF` which is in the package `reader`.

Reply:
Hmm, I think a malicious actor with access to any source code in the collector could cause some trouble with this. 

How about moving it to an internal package e.g. `fileconsumer/internal/time`? This way at least it's constrained to local code.

Reply:
I see the point and agree on not making any sec compromise here.

The thing is that [`Func`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.105.0/pkg/stanza/flush/flush.go#L29) function is part of the `stanza/flush` package while the flaky test `TestFlushPeriodEOF` is part of the `stanza/fileconsumer/internal/reader` package. So getting access from one package to the other in order to "mock" the var would require access to it anyways 🤔 ?

One maybe better way to achieve this would be to pass down to the `Func` the `time.Now` function so as to get access to it from the caller?

Any other suggestions more than welcome.

Reply:
Right 👍🏽 , that work. It didn't occur to me to go to the direction of a `pkg/stanza/internal`.